src/stores/AccountController.ts

```diff
@@ -229,67 +233,81 @@ export const updatePersonalShelves = async () => {
+      const response = await accountService.getCustomerConsents({ config, customer, jwt });
 
-    if (response && !response.errors?.length) {
-      useAccountStore.setState({ customerConsents: response.responseData.consents });
-    }
+      if (response && !response.errors?.length) {
+        useAccountStore.setState({ customerConsents: response.responseData.consents });
+      }
 
-    return response;
+      return response;
+    });
   });
 }
 
-export async function getPublisherConsents() {
-  return await useConfig(async ({ cleengId, cleengSandbox }) => {
-    const response = await cleengAccountService.fetchPublisherConsents({ publisherId: cleengId }, cleengSandbox);
+export const getPublisherConsents = async (): Promise<ServiceResponse<GetPublisherConsentsResponse>> => {
+  return await withAccountService(async ({ accountService, config }) => {
+    const response = await accountService.getPublisherConsents(config);
 
-    if (response && !response.errors?.length) {
-      useAccountStore.setState({ publisherConsents: response.responseData.consents });
-    }
+    useAccountStore.setState({ publisherConsents: response.responseData.consents });
 
     return response;
   });
-}
-
-export const getCaptureStatus = async () => {
-  return await useLoginContext(async ({ cleengSandbox, customerId, auth: { jwt } }) => {
-    const response = await cleengAccountService.getCaptureStatus({ customerId }, cleengSandbox, jwt);
+};
 
-    if (response.errors.length > 0) throw new Error(response.errors[0]);
+export const getCaptureStatus = async (): Promise<GetCaptureStatusResponse> => {
+  return await useAccountContext(async ({ customer, auth: { jwt } }) => {
+    return await withAccountService(async ({ accountService, sandbox }) => {
+      const { responseData } = await accountService.getCaptureStatus({ customer }, sandbox, jwt);
 
-    return response.responseData;
+      return responseData;
+    });
   });
 };
 
-export const updateCaptureAnswers = async (capture: Capture) => {
-  return await useLoginContext(async ({ cleengSandbox, customerId, auth }) => {
-    const response = await cleengAccountService.updateCaptureAnswers({ customerId, ...capture }, cleengSandbox, auth.jwt);
+export const updateCaptureAnswers = async (capture: Capture): Promise<Capture> => {
+  return await useAccountContext(async ({ customer, auth }) => {
+    return await withAccountService(async ({ accountService, accessModel, sandbox }) => {
+      const response = await accountService.updateCaptureAnswers({ customer, ...capture }, sandbox, auth.jwt);
 
-    if (response.errors.length > 0) throw new Error(response.errors[0]);
+      if (response.errors.length > 0) throw new Error(response.errors[0]);
 
-    // @todo why is this needed?
-    await getAccount(auth);
+      // if no refresh token present (InPlayer config), update account store
+      // otherwise fetch account
+      if (!auth.refreshToken) {
```

Comment:
If you add the query from getAccount (service.getUser) into the cleeng implementation of updateCaptureAnswers then you don't need this condition here, you can just always call afterLogin.